app/controllers/submissions_controller.rb

```diff
@@ -11,9 +11,8 @@ def create
     @task.update status: :waiting if @task.not_submitted? || @task.waiting?
     @submission = Submission.create(submission_params)
     UserMailer.new_submission_notify(@submission).deliver
+    Notification.new_event(user: @submission.teacher, task: @submission.task, event_type: :new_submission, event_time: @submission.created_at)
```

Comment:
But the `task.updated_at` is also updated on each note/submission creation. You can use it in `Notification.new_event`.